src/messaging/ReliableMessageManager.cpp

```diff
@@ -482,21 +499,18 @@ void ReliableMessageManager::ClearRetransmitTable(ReliableMessageContext * rc)
  *  @param[in]    rEntry   A reference to the RetransTableEntry object.
  *
  */
-void ReliableMessageManager::ClearRetransmitTable(RetransTableEntry & rEntry)
+void ReliableMessageManager::ClearRetransTable(RetransTableEntry & rEntry)
 {
     if (rEntry.rc)
     {
+        VerifyOrDie(rEntry.rc->mExchange != nullptr);
+
         // Expire any virtual ticks that have expired so all wakeup sources reflect the current time
         ExpireTicks();
 
-        rEntry.rc->Release();
+        rEntry.rc->mExchange->Release();
```

Comment:
The life-cycle of exchange is co-owned by ReliableMessageManager and protocol, rc is part of exchange.

Reply:
The fact that `rc` is part of exchange seems like an implementation detail from the point of view of this code.  This code has `rc`.  It should manage the lifetime of the thing it points to.  The fact that this happens to be implemented via merely keeping the exchange alive is an implementation detail of `ReliableMessageContext`....

If we were storing the _exchange_ in the table, and recovering the `ReliableMessageContext` from that as needed, I would be all in favor of directly managing the exchange lifetime.  Or put another way, we should structure this such that if we ever decide to use a smart pointer for the lifetime management it will Just Work instead of having to rejigger things; that's the right mental model.

Reply:
Acked